contacts2/app/controllers/contacts_controller.rb

```diff
@@ -0,0 +1,54 @@
+class ContactsController < ApplicationController
+    def home
+    end
+    
+    def index
+        @contacts = Contact.all
+    end
+    
+    def new
+    end
+    
+    def create
+        contact = Contact.new
+        contact.first_name = params[:first_name]
+        contact.last_name = params[:last_name]
+        contact.phone = params[:phone]
+        contact.email = params[:email]
+        contact.address = params[:address]
+        contact.city = params[:city]
+        contact.state = params[:state]
+        contact.zip = params[:zip]
+        contact.save
+        redirect_to '/contacts'
+    end
+    
+    def edit
+        @edit = Contact.find_by(id: params[:id])
+    end
+    
+    def update
+        contact = Contact.find_by(id: params[:id])
+        contact.first_name = params[:first_name]
+        contact.last_name = params[:last_name]
+        contact.phone = params[:phone]
+        contact.email = params[:email]
+        contact.address = params[:address]
+        contact.city = params[:city]
+        contact.state = params[:state]
+        contact.zip = params[:zip]
+        contact.save
+        redirect_to '/contacts'
```

Comment:
And here you can also resuse the `contact_params` method and do something like 
`    if contact.update_attributes(contact_params)
          redirect_to contacts_path
        end` 
`update_attributes` basically allows to update any attribute in a contact and saves the it. You can read the [docs](https://api.rubyonrails.org/classes/ActiveRecord/Persistence.html#method-i-update_attributes) for more details